docs/faq.rst

```diff
@@ -0,0 +1,37 @@
+Frequently Asked Questions (FAQ)
+================================
+
+This is a collection of answers to frequently asked questions (FAQ). You can contribute to the Flax FAQ by starting a new topic in `GitHub Discussions <https://github.com/google/flax/discussions>`__.
+
+Where to search for an answer to a Flax-related question
+********************************************************
+
+There are a number of official Flax resources to search for information:
+
+- `Flax Documentation on ReadTheDocs <https://github.com/google/flax/discussions>`__ (this site): Use the search bar or the table of contents on the left-hand side.
+- `google/flax GitHub Discussions <https://github.com/google/flax/discussions>`__: Search for an existing topic or start a new one. If you can't find what you're looking for, feel free to ask the Flax team or community a question.
+- `google/flax GitHub Issues <https://github.com/google/flax/discussions>`__.
+
+How to take the derivative with respect to a hidden layer values (using :code:`Module.perturb`)
+***********************************************************************************************
+
+To take the derivative(s) or gradient(s) of the output with respect to a hidden/intermediate activation inside a model layer, you can use :meth:`flax.linen.Module.perturb`. You define a zero-value :class:`flax.linen.Module` "perturbation" parameter – :code:`perturb(...)` – in the forward pass with the same shape as the intermediate activation, define the loss function with :code:`'perturbations'` as an added standalone argument, perform a JAX derivative operation with :code:`jax.grad` on the perturbation argument.
+
+For full examples and detailed documentation, go to:
+
+- The :meth:`flax.linen.Module.perturb` API docs
+- The `Extracting gradients of intermediate values <https://flax.readthedocs.io/en/latest/guides/extracting_intermediates.html#extracting-gradients-of-intermediate-values>`_ guide
+- `Flax GitHub Discussions #1152 <https://github.com/google/flax/discussions/1152>`_
+
+Is Flax Linen :code:`remat_scan()` the same as :code:`scan(remat(...))`?
+************************************************************************
+
+Flax :code:`remat_scan()` (:meth:`flax.linen.remat_scan()`) and :code:`scan(remat(...))` (:meth:`flax.linen.scan` over :meth:`flax.linen.remat`) are not the same, and :code:`remat_scan()` is limited in cases it supports.
```

Comment:
Here we should clarify what "is limited in cases it supports" means, specifically that `remat_scan` treats the inputs and outputs as carries (there are no scan dimension for the data). This makes it easy to implement the "scan over layers" pattern for an existing Module as shown in the docs for `remat_scan`, with `scan(remat(...))` you have create a wrapper Module to handle the scan inputs/outputs.